### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Goooler**
- Extract strings to `libs.versions.toml`.
- Moshi 1.12.0
- Guava 30.1.1-jre
- Ktlint 0.42.1

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -24,8 +24,7 @@ import java.net.URL
 buildscript {
   dependencies {
     classpath(kotlin("gradle-plugin", version = libs.versions.kotlin.get()))
-    // https://github.com/melix/japicmp-gradle-plugin/issues/36
-    classpath("com.google.guava:guava:28.2-jre")
+    classpath(libs.guava)
```
Comment (ZacSweers) [inline build.gradle.kts:27]: Please restore the comment

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -45,6 +45,9 @@ kotlinpoet-metadata-core = { module = "com.squareup:kotlinpoet-metadata", versio
 kotlinpoet-metadata-specs = { module = "com.squareup:kotlinpoet-metadata-specs", version.ref = "kotlinpoet" }
 kotlinxMetadata = "org.jetbrains.kotlinx:kotlinx-metadata-jvm:0.3.0"
 okio = "com.squareup.okio:okio:2.10.0"
+guava = "com.google.guava:guava:30.1.1-jre"
+moshi = { module = "com.squareup.moshi:moshi", version.ref = "moshi" }
+moshiAdapters = { module = "com.squareup.moshi:moshi-adapters", version.ref = "moshi" }
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:50]: Let's call these something more specific to indicate they are just for japicmp. Same with the moshi version itself

File: build.gradle.kts (hunk 1 above)
Comment (Goooler) [reply build.gradle.kts:28]: Rverted.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (Goooler) [reply gradle/libs.versions.toml:50]: What do you mean ? Can you just give me suggest changes ?

**Comment by ZacSweers** [comment]
After thinking on this more, I'm going to close this for the same reason as I did your [EitherNet PR](https://github.com/slackhq/EitherNet/pull/35#issuecomment-917353636). Unless you use Moshi and have a specific use case for these updates, please don't send these types of PRs just to fluff your contribution stats.

**Pull request closed without merging**